src/components/AccountModule/AccountModule.js

```diff
@@ -1,78 +1,49 @@
-import React, { useRef } from 'react'
+import React, { useRef, useState, useCallback } from 'react'
 import PropTypes from 'prop-types'
 import {
-  Button,
   EthIdenticon,
   Popover,
   GU,
   IconDown,
-  IconConnect,
   RADIUS,
   textStyle,
   useTheme,
-  IdentityBadge,
   unselectable,
+  ButtonBase,
 } from '@aragon/ui'
 import { shortenAddress } from '../../web3-utils'
 import { useAccount } from '../../account'
-
-function getNetworkName(networkId) {
-  if (networkId === 'main') return 'Mainnet'
-  if (networkId === 'rinkeby') return 'Rinkeby'
-  return networkId
-}
+import NotConnected from './NotConnected'
+import ConnectionInfo from './ConnectionInfo'
+import { useConnectionColor, useNetworkConnectionData } from './utils'
 
 function AccountModule({ compact }) {
   const { connected } = useAccount()
-  return connected ? <ConnectedMode /> : <NonConnectedMode compact={compact} />
+  return connected ? <ConnectedMode /> : <NotConnected compact={compact} />
```

Comment:
It'd be nice to have a loading state for this, so we don't see the initial flicker with the not connected button

Reply:
Yes! I added a delay rather than a loader, so that we avoid another flicker :laughing: 

https://github.com/aragon/aragon/pull/1137/commits/fac3e496c38823af4854a05155885443af675c60 and https://github.com/aragon/aragon/pull/1137/commits/31e31e83af754e1402740d729eb395fa9c597fa0